client/client.go

```diff
@@ -247,7 +249,10 @@ func (c *Client) downloadTimestamp() error {
 	raw, err := c.remote.GetMeta(role, maxSize)
 	var s *data.Signed
 	if err != nil || len(raw) == 0 {
-		if err, ok := err.(*store.ErrMetaNotFound); ok {
+		if err, ok := err.(store.ErrMetaNotFound); ok {
+			return err
+		}
+		if old == nil {
 			return err
```

Comment:
This doesn't look quite right. Either the `len(raw) == 0`check is wrong, or we should be doing something other than `return err` here.